ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py

```diff
@@ -795,9 +795,10 @@ def test_swapaxes(
 
 # clip
 @given(
+    # use filter to ensure dtypes are all the same
     x_min_n_max=helpers.dtype_and_values(
         available_dtypes=ivy_np.valid_numeric_dtypes, num_arrays=3
-    ),
+    ).filter(lambda x_min_n_max: all(x == x_min_n_max[0][0] for x in x_min_n_max[0])),
```

Comment:
if we want the `dtype` to all be the same, then you should just pass `shared_dtype=True` into `dtype_and_values`.

Reply:
Yeah, you are right! 
The filter is superfluous :D